lib/MongoDB/Role/_CommandCursorOp.pm

```diff
@@ -46,6 +46,10 @@ sub _build_result_from_cursor {
         $self->cursorType eq 'tailable_await') {
         $max_time_ms = $self->maxAwaitTimeMS if $self->maxAwaitTimeMS;
     }
+    elsif ($self->isa('MongoDB::Op::_Aggregate') &&
+        $self->cursorType eq 'tailable_await') {
```

Comment:
As I said above, I think it's sufficient to just check if isa MongoDB::Op::_Aggregate (and that maxAwaitTimeMS is true) without an extra cursor_type.